### DISCUSSION TITLE
How to use FP16 model precision for inference ?

### DISCUSSION THREAD

**Discussion by jerin-metrum-ai** [openai/whisper#1835] [General]
I believe the default model precision of whisper models if FP32. so how can i use FP16 model using openai-whisper package.

**Comment by glangford** [comment]
`--fp16 True`

fp16 is true by default. See `whisper --help`.

**Reply by jerin-metrum-ai** [reply]
Thanks. I am using python openai-whisper package for loading and transcribing the audio. The --fp16 flag is part of transcribe() but the actual model is loaded into device using load_model().

**Reply by glangford** [reply]
Right. As @phineas-pta points out, the audio is loaded in fp16 not the model (and presumably mixed precision is used by pytorch).

There is no parameter for fp16 in `load_model`, see here
https://github.com/openai/whisper/blob/e58f28804528831904c3b6f2c0e473f346223433/whisper/__init__.py#L99

**Reply by jerin-metrum-ai** [reply]
Got it. 
I was printing model dtype and i could see the all the layers with float32 as shown below. I believe the default precision is Float32 

Parameter: encoder.conv1.weight, dtype: torch.float32
Parameter: encoder.conv1.bias, dtype: torch.float32
Parameter: encoder.conv2.weight, dtype: torch.float32
Parameter: encoder.conv2.bias, dtype: torch.float32
Parameter: encoder.blocks.0.attn.query.weight, dtype: torch.float32
Parameter: encoder.blocks.0.attn.query.bias, dtype: torch.float32
Parameter: encoder.blocks.0.attn.key.weight, dtype: torch.float32
Parameter: encoder.blocks.0.attn.value.weight, dtype: torch.float32
Parameter: encoder.blocks.0.attn.value.bias, dtype: torch.float32
Parameter: encoder.blocks.0.attn.out.weight, dtype: torch.float32
Parameter: encoder.blocks.0.attn.out.bias, dtype: torch.float32
Parameter: encoder.blocks.0.attn_ln.weight, dtype: torch.float32

**Reply by ex3ndr** [reply]
tiny one is float16 though

**Comment by phineas-pta** [comment]
`fp16` is a misleading option, it loads the audio in float16 not the model, see #622

for truly float16 use other interfaces like `transformers` `whisper.cpp` `faster-whisper`

**Reply by jerin-metrum-ai** [reply]
Thanks for the response. I will try out some of the suggested options.
